packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2646,29 +2647,25 @@ export class ContainerRuntime
 		if (hasModernRuntimeMessageEnvelope) {
 			// If the message has the modern message envelope, then process it here.
 			// Here we unpack the message (decompress, unchunk, and/or ungroup) into a batch of messages with ContainerMessageType
-			const processResult = this.remoteMessageProcessor.process(messageCopy, logLegacyCase);
-			if (processResult === undefined) {
+			const inboundBatch = this.remoteMessageProcessor.process(messageCopy, logLegacyCase);
+			if (inboundBatch === undefined) {
 				// This means the incoming message is an incomplete part of a message or batch
 				// and we need to process more messages before the rest of the system can understand it.
 				return;
 			}
-			const batchStartCsn = processResult.batchStartCsn;
-			const batch = processResult.messages;
-			const sequenceNumber = processResult.sequenceNumber;
+
 			const messages: {
 				message: InboundSequencedContainerRuntimeMessage;
 				localOpMetadata: unknown;
 			}[] = local
 				? this.pendingStateManager.processPendingLocalBatch(
-						batch,
-						batchStartCsn,
-						sequenceNumber,
+						inboundBatch.messages,
+						inboundBatch.batchStartCsn,
+						inboundBatch.emptyBatchSequenceNumber,
```

Comment:
No reason to send it completely for now but I guess eventually we want to send the entire inboundBatch

Reply:
Yeah  minimizing conflict with my other branch